File: java/iroha_android/src/main/java/org/hyperledger/iroha/android/crypto/export/KeyPassphraseProvider.java
package org.hyperledger.iroha.android.crypto.export;

/**
 * Supplies passphrases for deterministic key export/import flows.
 *
 * <p>Implementations should return a fresh char array per invocation so callers can safely zero
 * the returned buffer after use.
 */
@FunctionalInterface
public interface KeyPassphraseProvider {
  /** Returns the current passphrase as a mutable character array. */
  char[] passphrase();
}

File: kotlin/core-jvm/src/main/java/org/hyperledger/iroha/sdk/crypto/export/KeyPassphraseProvider.kt
package org.hyperledger.iroha.sdk.crypto.export

/**
 * Supplies passphrases for deterministic key export/import flows.
 *
 * Implementations should return a fresh char array per invocation so callers can safely zero
 * the returned buffer after use.
 */
fun interface KeyPassphraseProvider {
    /** Returns the current passphrase as a mutable character array. */
    fun passphrase(): CharArray
}
